internal/util/pkgutil/pkgutil.go

```diff
@@ -15,25 +15,67 @@
 package pkgutil
 
 import (
-	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 
 	kptfilev1alpha2 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1alpha2"
 	"github.com/GoogleContainerTools/kpt/pkg/kptfile/kptfileutil"
 	"sigs.k8s.io/kustomize/kyaml/copyutil"
 )
 
-type RemoteSubPkgInfo struct {
-	PackagePath          string
-	DeclaringKptfilePath string
+// FindAllDirectSubpackages returns the paths to all immediate subpackages of the
+// provided path. It finds both remote and local subpackages, but it will not
```

Comment:
What do you mean by remote subpackages ? Isn't that concept going away?

Reply:
We still use the distinction between local and remote packages, which is determined by whether the Kptfile has an `upstream` section defined. During updates, we do want to update subpackages, and without a list in the parent, we find those packages by scanning the filesystem.